src/test/testUtils.tsx

```diff
@@ -0,0 +1,46 @@
+import * as rtl from "@testing-library/react";
+import type { RenderOptions } from "@testing-library/react";
+import React, { PropsWithChildren } from "react";
+import { Provider } from "react-redux";
+import type { PreloadedState } from "@reduxjs/toolkit";
+import type { UIState } from "ui/state";
+import type { UIStore } from "ui/actions";
+
+import { bootstrapStore, extendStore } from "ui/setup/store";
+import setupDevtools from "ui/setup/dynamic/devtools";
+// import type { AppStore, RootState } from "../store";
+
+// This type interface extends the default options for render from RTL, as well
+// as allows the user to specify other things such as initialState, store. For
+// future dependencies, such as wanting to test with react-router, you can extend
+// this interface to accept a path and route and use those in a <MemoryRouter />
+interface ExtendedRenderOptions extends Omit<RenderOptions, "queries"> {
+  preloadedState?: Partial<UIState>;
+  store?: UIStore;
+}
+
+export async function createTestStore(preloadedState: Partial<UIState> = {}) {
```

Comment:
Would it make sense to consider `DeepPartial`? There probably is some value in that proposal, but considering how we generally tended to organize store so far (pretty flat), maybe not so.

Reply:
No, you really _can't_ do `DeepPartial` - each slice reducer either gets all of its slice state or `undefined`, so it can't be "deep"